micall/tests/test_contig_stitcher.py

```diff
@@ -0,0 +1,804 @@
+                        match_fraction=0.5,
+                        ),
+        ]
+
+    results = list(stitch_contigs(contigs))
+    assert len(results) == 1
+
+    result = results[0]
+
+    assert 100 == len(result.seq)
+    assert result.seq == 'A' * 50 + 'C' * 50
+    assert result.query.name == 'left(a)+overlap(a,b)+right(b)'
+
+
+def test_correct_processing_of_two_overlapping_and_one_separate_contig(exact_aligner):
+    # Scenario: Two overlapping contigs are stitched together, the non-overlapping is kept separate.
+
+    ref_seq = 'A' * 100 + 'C' * 100 + 'T' * 100
+
+    contigs = [
+        GenotypedContig(name='a',
+                        seq='A' * 50 + 'C' * 20,
+                        ref_name='testref',
+                        ref_seq=ref_seq,
+                        match_fraction=0.5,
+                        ),
+        GenotypedContig(name='b',
+                        seq='A' * 20 + 'C' * 50,
+                        ref_name='testref',
+                        ref_seq=ref_seq,
+                        match_fraction=0.5,
+                        ),
+        GenotypedContig(name='c',
+                        seq='C' * 20 + 'T' * 50,
+                        ref_name='testref',
+                        ref_seq=ref_seq,
+                        match_fraction=0.5,
+                        ),
+        ]
+
+    results = list(stitch_contigs(contigs))
+    assert len(results) == 2
+
+    assert 100 == len(results[0].seq)
+    assert results[0].seq == 'A' * 50 + 'C' * 50
+    assert results[0].query.name == 'left(a)+overlap(a,b)+right(b)'
+
+    assert results[1].query == contigs[2]
+
+
+def test_stitching_of_all_overlapping_contigs_into_one_sequence(exact_aligner):
+    # Scenario: All contigs have some overlapping parts, resulting in one continuous sequence after stitching.
+
+    ref_seq = 'A' * 100 + 'C' * 100 + 'T' * 100
+
+    contigs = [
+        GenotypedContig(name='a',
+                        seq='A' * 50 + 'C' * 20,
+                        ref_name='testref',
+                        ref_seq=ref_seq,
+                        match_fraction=0.5,
+                        ),
+        GenotypedContig(name='b',
+                        seq='A' * 20 + 'C' * 100 + 'T' * 20,
+                        ref_name='testref',
+                        ref_seq=ref_seq,
+                        match_fraction=0.5,
+                        ),
+        GenotypedContig(name='c',
+                        seq='C' * 20 + 'T' * 50,
+                        ref_name='testref',
+                        ref_seq=ref_seq,
+                        match_fraction=0.5,
+                        ),
+        ]
+
+    results = list(stitch_contigs(contigs))
+    assert len(results) == 1
+
+    result = results[0]
+
+    assert 200 == len(result.seq)
+    assert result.seq == 'A' * 50 + 'C' * 100 + 'T' * 50
+    assert result.query.name == 'left(a)+overlap(a,b)+left(right(b))+overlap(left(a)+overlap(a,b)+right(b),c)+right(c)'
```

Comment:
I think this is too complicated for users to parse. What about `'a, a/b, b, b/c, c'`? Remember also that the original contig names are pretty long. I think they're all a number prefix added to the accession number that Blast found.

Reply:
These names are not returned to users, instead each contig is named like `contig-$N-$SUFFIX_BASED_ON_REF_NAME`. So would it be okay to leave these long names here for debugging purposes?